src/ai2_internal/citation_mentions/interface.py

```diff
@@ -39,6 +39,34 @@ class PredictorConfig(BaseSettings):
     pass
 
 
+def group_by_line(boxes):
+    boxes = sorted(boxes, key=lambda box: box.t)
+    return [list(line_boxes) for t, line_boxes in groupby(boxes, key=lambda box: box.t)]
```

Comment:
Are you seeing that the line grouping is working well? I think it could be better if we had `rows` passed into this model, so that we can take advantage of pdfplumber's grouping by lines which I think is more robust than requiring exact `t` coordinates matching up (see https://github.com/allenai/mmda/blob/5c2fb76a3ec017f20f27dca056ce57a425961296/src/mmda/parsers/pdfplumber_parser.py#L370) -- or maybe just box.t +/- a little amount calculated from getting average token heights? (see https://github.com/allenai/mmda/blob/6ab9004e0df5466201396163da8be6c34e222d2d/src/mmda/predictors/heuristic_predictors/figure_table_predictors.py#L53) -- just options, but if you're seeing good results with this grouping I wouldn't worry about making this more complicated

Reply:
it actually seems to work well! if the lines dont have the same `t` then I think it's safest to just not do anything.